tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/impl/MetricsHelper.java

```diff
@@ -0,0 +1,236 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.framework.metrics.impl;
+
+import org.tensorflow.Operand;
+import org.tensorflow.ndarray.Shape;
+import org.tensorflow.op.Op;
+import org.tensorflow.op.Ops;
+import org.tensorflow.op.math.Mean;
+import org.tensorflow.types.TBool;
+import org.tensorflow.types.TFloat32;
+import org.tensorflow.types.TInt32;
+import org.tensorflow.types.family.TNumber;
+import org.tensorflow.types.family.TType;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.tensorflow.framework.losses.impl.LossesHelper.allAxes;
+import static org.tensorflow.framework.utils.CastHelper.cast;
+
+/**
+ * These are helper methods for Metrics and will be module private when Java modularity is applied
+ * to TensorFlow Java. These methods should not be used outside of the metrics packages.
+ */
+public class MetricsHelper {
+  public static final float NEG_INF = -1e10f;
+  private static final String ASSERT_BROADCAST_ERROR_PREFIX =
+      "weights can not be broadcast to values.";
+
+  /**
+   * Asserts that the <code>sampleWeight</code> can be broadcast to <code>values</code>
+   *
+   * @param tf the TensorFlow Ops
+   * @param sampleWeights the sample weights.
+   * @param values the values to which weights are applied.
+   * @return <code>Operation</code> raising <code>InvalidArgumentError</code> if <code>sampleWeight
+   *     </code> has incorrect shape. <code>no_op</code> if static checks determine <code>
+   *     sampleWeight</code> has correct shape.
```

Comment:
we shouldn't document the thrown exceptions in the `@return` clause but in `@throws` (which is already done)

Reply:
Fixed.